### PR TITLE
Vargac/vic intern functions

### PR DESCRIPTION
**Pull request by vargac**
* We have only two intern functions right now: set, get
* Intern funcs are similar to user-defined ones. They are also stored in the
  same array, but user doesn't know about them.
* Definitions of intern funcs are in stdlib
* If user defines function with the same name, it will be called only, if
  intern one will throw any error
* User can set how output data will be processed. Printed to terminal, stored
 in buffer, etc. He tells to vic which function does it (by vic_ouput_set), and
 then vic will send character by character through that function.
* Three outputing functions:
    * vic_out - outputs character
    * vic_print - outputs string
    * vic_println - vic_print with new line at the end
    * vic_print_err - outputs error message

### CODE REVIEW

File: src/vic-stdlib.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+#include "vic.h"
+#include "stdlib.h"
+
+void vic_set(void)
+{
+    char var_name[VIC_VAR_NAME_LEN+1] = {'\0'};
+    char var_val[VIC_VAR_VAL_LEN+1] = {'\0'};
+    char* args_format = \
+        "%"VIC_XSTR(VIC_VAR_NAME_LEN)"s %"VIC_XSTR(VIC_VAR_VAL_LEN)"s";
```
Comment (mrshu) [inline src/vic-stdlib.c]: @vargac this looks pretty nasty but I do not really see a nicer way of doing this right away.

**Comment by mrshu** [comment]
@vargac First of all, thanks a ton for this!

What I am not sure I like is the way you decided to handle the internal functions and their rebinding. Is that really necessary? What is the added value of separating the internal functions from the others?

File: src/vic-stdlib.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    if (vic_args(args_format, var_name) != 1) {
+        vic_print_err(VIC_ERR_INVALID_ARGS);
+        return;
+    }
+
+    char *value = NULL;
+    int8_t error = vic_var_get(var_name, &value);
+
+    vic_print_err(error);
+    if (value != NULL) {
```
Comment (mrshu) [inline src/vic-stdlib.c]: @vargac if you had to add this check I believe it begs for a test =)

File: tests/test-vic-exec.c [hunk 3]
```diff
@@ -91,12 +91,109 @@ static char * test_exec_bad_par_func_2(void)
...
+static char *test_exec_set_get(void)
+{
+    int test_count = 5;
+
+    char *in[] = {
+        " set name Peter \n",
+        "  set age 7 \n",
+        " get    \n",
+        "get age \n",
+        " get name\n"
```
Comment (mrshu) [inline tests/test-vic-exec.c]: @vargac I believe you miss the test for getting a variable that does not exist.

File: Makefile [hunk 4]
```diff
@@ -1,6 +1,7 @@
 CC=gcc -g -Wall
 OFLAGS=-lm
-OBJ=src/vic.o src/vic-serial.o src/vic-funcs.o src/vic-exec.o src/vic-var.o
+OBJ=src/vic.o src/vic-serial.o src/vic-funcs.o src/vic-exec.o src/vic-var.o \
```
Comment (mrshu) [inline Makefile]: @vargac we need to get rid of these things and figure out how to structure the Makefile better.

File: src/vic-funcs.c [hunk 5]
```diff
@@ -58,10 +64,27 @@ int8_t vic_fn_call(const char *raw_name)
...
+            return VIC_ERR_NO;
+        }
+    }
+    return VIC_ERR_INVALID_NAME;
+}
+
 int8_t vic_fn_rm(const char *raw_name)
 {
     char name[VIC_FUNC_NAME_LEN + 1];
-    vic_prepare_name(raw_name, name, VIC_FUNC_NAME_LEN);
```
Comment (mrshu) [inline src/vic-funcs.c]: @vargac can this function be removed?

File: src/vic-var.h [hunk 6]
```diff
@@ -5,7 +5,7 @@
 
 struct vic_var_t {
     char name[VIC_VAR_NAME_LEN + 1];
-    char *p_value;
+    char value[VIC_VAR_VAL_LEN+1];
```
Comment (mrshu) [inline src/vic-var.h]: @vargac Can we add spaces here around `+`?

File: src/vic-funcs.c (hunk 5 above)
Comment (vargac) [inline src/vic-funcs.c]: @mrshu Yes, we use strncpy now

File: Makefile (hunk 4 above)
Comment (vargac) [inline Makefile]: @mrshu I have not seen many makefiles yet, so I worry I don't know how to do it. But if we are going to improve makefiles, I suggest also to look at tests/Makefile, when running all-tests, I see 'All passed' even when some don't pass.

**Comment by vargac** [comment]
@mrshu I decided to separate internal functions, in order to hide them from user. If user sets VIC_FUNCS_COUNT to 64, he will get array for 64 vic-funcs and if he look at first element of this array, he will see what he expects. Also, user may want to call his vic-function by vic_fn_call, but without knowledge of intern vic-funcs, he might get strange results. Actually, they are stored in the same array, but only pointer to start of user-defined vic-funcs region is shown to world outside vic.

**Comment by mrshu** [comment]
@vargac nice job, :+1:

**Pull request merged**